src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs

```diff
@@ -219,7 +221,7 @@ private static List<int> GetCategoricalBoundires(int[] categoricalFeatures, int
         private static List<string> ConstructCategoricalFeatureMetaData(int[] categoricalFeatures, int rawNumCol, ref CategoricalMetaData catMetaData)
         {
             List<int> catBoundaries = GetCategoricalBoundires(categoricalFeatures, rawNumCol);
-            catMetaData.NumCol = catBoundaries.Count() - 1;
```

Comment:
>Count() [](start = 47, length = 7)

Ah very good. Yes that is terrible. There really be a warning or something on `Count()`... :P